src/torrent.cpp

```diff
@@ -1770,7 +1770,7 @@ namespace libtorrent
 			// maybe this whole section should move to need_picker()
 			need_picker();
 
-			peer_request pr = m_torrent_file->map_file(i, 0, fs.file_size(i));
+			peer_request pr = m_torrent_file->map_file(i, 0, int(fs.file_size(i)));
```

Comment:
I see now the reason, `peer_request::length` can't be that long. No problem in these cases since offset is `0` (just looking for the first `peer_request`).